### ISSUE TITLE
Only English keyboard shortcuts are used 

### ISSUE THREAD

**Issue by RC2225** [luciusDXL/TheForceEngine#212]
Different languages use different highlighted characters to navigate the menu. As an example in German Ja (yes) it would be J and not Y. 

![](https://user-images.githubusercontent.com/5329274/216113797-c343e96b-b2d5-4b41-b416-f3a144646dbb.png)

**Comment by TP555** [comment]
Hi

Yes , still not fixed  !

**Comment by mlauss2** [comment]
Hm, we could add language-specific keymappings for this dialog, and use e.g. LOCAL.MSG message 560 or 600 as to distinguish between them?

**Comment by mlauss2** [comment]
The easiest fix is to simply make 'J' equivalent to 'Y':

```diff
--- a/TheForceEngine/TFE_DarkForces/GameUI/escapeMenu.cpp
+++ b/TheForceEngine/TFE_DarkForces/GameUI/escapeMenu.cpp
@@ -572,7 +572,7 @@ namespace TFE_DarkForces
                {
                        if (actionPressed < 0)
                        {
-                               if (TFE_Input::keyPressed(KEY_Y))
+                               if (TFE_Input::keyPressed(KEY_Y) || TFE_Input::keyPressed(KEY_J))
                                {
                                        actionPressed = CONFIRM_YES;
                                }


```

**Comment by RC2225** [comment]
List from every menu entry where nothing is changed from english is empty.  
|ENG|GER|FRE|ITA|SPA|
|---|---|---|---|---|
|Yes (Y)|J|O|S|S|
|No (N)|||||
|New Agent (N)|||||
|Remove Agent (R)|L|E||E|
|DOS (D)|||||
|Start Mission (S)||C|I|I|
|Easy (E)|L|F|F|F|
|Medium (M)||N|||
|Hard (H)||D|D|D|
|Cancel (C)|A|A|A||
|OK (O)|||||
|Abort mission (A)|||||
|Configration (C)|K||||
|Quit to DOS (Q)|B|D|E|L|
|Return to Game (R)|F|||


Images: 
[menu_images.zip](https://github.com/luciusDXL/TheForceEngine/files/10578994/menu_images.zip)

**Comment by mlauss2** [comment]
Thanks!  I'm currently working on implementing per-language keybindings for the menus.

**Comment by mlauss2** [comment]
btw, can you also provide the local.msg string 600 for italian and spanish?  Or do you know of a realiable way to detect the language of a particular DF installation?  Currently I'm looking for a custom string in local.msg, but am open to more robust methods.
Danke!

**Comment by mlauss2** [comment]
Try #217 please.   I tested english and german versions, they work now as intended.

**Comment by RC2225** [comment]
Italian: 
`600 0:  "Non hai abbastanza memoria RAM per far girare Dark Forces.\nLeggi il manuale per creare un Boot Disk."`
Spanish:
`600 0:  "No tienes suficiente memoria RAM disponible para ejecutar Dark Forces.\nMira el manual para hacer un Disco de Arranque."` 
French:
`600 0:  "No tienes suficiente memoria RAM disponible para ejecutar Dark Forces.\nMira el manual para hacer un Disco de Arranque."`

I assumed they used different exes between the language version but they are, at least on Steam, identical. The navigation seems to be defined in the DARK.GOB under MSGS 33. According to the Spec 33 only indicates that there are 33 Msgs. 

https://df-21.net/documentation/DFSPECS/df_specs.htm

**Comment by mlauss2** [comment]
I've update PR #217 with parsing of the HOTKEYS.MSG file, please give it a try, it works on my english and german versions.

**Comment by RC2225** [comment]
Tried with french, italian and spanish and works. I think if tested every combination. :)

**Comment by TP555** [comment]
Hi

Fixed now in 1.08 for the German Version so far
